pkgs/development/misc/stm32/betaflight/default.nix

```diff
@@ -21,8 +19,8 @@ in stdenv.mkDerivation rec {
   src = fetchFromGitHub {
     owner = "betaflight";
     repo = "betaflight";
-    rev = "v${version}";
-    sha256 = "0vbjyxfjxgpaiiwvj5bscrlfikzp3wnxpmc4sxcz5yw5mwb9g428";
+    rev = "041362614bac4aaed4c0e4cc163a1ba99139dfdc";
```

Comment:
Why stop using the version?

Reply:
It's because of the first line in `postPatch`; the short form of the revision is baked into the firmware images. If I use the tag, I will still need to store the commit id someplace, unless you know of a way to convert the tag to a commit id at build time, without pulling in any extra buildInputs. Is it cached someplace in the src derviation by the builder?

I'll add a note for now.

Reply:
I didn't notice it was being used there. I looked into it some, but it doesn't look like there is a better solution than what you have, unless `leaveDotGit` works here. It looks like `leaveDotGit` has some issues of its own though https://github.com/NixOS/nixpkgs/issues/8567#issuecomment-148918815